src/hip_hcc.cpp

```diff
@@ -1051,7 +1062,8 @@ void ihipCtx_t::locked_syncDefaultStream(bool waitOnSelf, bool syncHost) {
 
         if (HIP_SYNC_NULL_STREAM) {
             if (waitThisStream) {
```

Comment:
Should `waitThisStream` be the first order check instead of being nested inside?

Reply:
I don't disagree that the first order check might need revision, but not in this PR.